### PR TITLE
SEC-1586 fix: CmsConsentManager consent query missing tenant/owner discriminator

### PR DESCRIPTION
**Pull request by alvinhenrick**
Jira: https://icanbwell.atlassian.net/browse/SEC-1586

## What this fixes

`CmsConsentManager.getConsentResources` built its Consent query on status/proxy-patient/category/provision-type only — no owner/access tag filter — unlike the sibling `ProaConsentManager`, which already filters by `ownerTags`. A CMS partner could satisfy a consent requirement using another tenant's Consent resource for a patient already inside its own linked compartment.

## The fix

Mirror `ProaConsentManager`'s pattern: add an owner-tag `$elemMatch` to the query itself (the join condition, not a later filter step). Threaded `securityTags` (the caller's own tenant scope, already computed in `searchManager.js`) through `updateQueryConsideringCmsDataSharing` → `cmsConsentManager.getPatientIdsWithConsent` → `getConsentResources`.

## Adversarial review against `review.md`

Checked the empty-filter trap (section D): a caller with wildcard `access/*.*` scope resolves to `securityTags=[]`, making the owner filter's `$in:[]` match zero Consent resources — so the function correctly returns the "deny" result rather than falling through to "no filter, return everything." This mirrors `ProaConsentManager`'s existing behavior for the same edge case exactly — not a new asymmetry, and fail-closed is the right direction for a security fix.

## Verification

- 20 `cmsConsentManager` unit tests (5 new) pass.
- 1 new `dataSharingManager` test verifying `securityTags` threading passes.
- Full unit suite: identical 78 pre-existing failures before/after (tracked SEC-1582/DCON-4775 bug-documentation tests) — zero new regressions.

## Test plan

- [x] Unit tests pass
- [x] No new regressions in full unit suite
- [ ] Code-owner review (branch protection requires 1 approving review)

### CODE REVIEW

File: src/operations/search/cmsConsentManager.js [hunk 1]
```diff
@@ -39,7 +40,15 @@ class CmsConsentManager {
...
+                {
+                    'meta.security': {
+                        $elemMatch: {
+                            system: 'https://www.icanbwell.com/owner',
+                            code: { $in: ownerTags }
+                        }
+                    }
+                }
             ]
         };
```
[the remark this answers is not in the document]
Comment (alvinhenrick) [reply src/operations/search/cmsConsentManager.js:49]: Fixed in 7bf5e8967: `getConsentResources` now only adds the `$elemMatch` owner clause when `ownerTags` is non-empty. An empty array means the caller has wildcard (`*`) access — the same convention `searchManager.js`'s sibling `securityTags` branch already uses to skip its own tag filter — so it's treated as unrestricted rather than `$in: []`, which matched nothing. Added regression tests for both the empty-array and undefined cases in `cmsConsentManager.test.js`.

**Review by mintusah25 — APPROVED** [review]
